sources/packages/backend/apps/api/src/route-controllers/institution/_tests_/e2e/institution.aest.controller.searchInstitutions.e2e-spec.ts

```diff
@@ -0,0 +1,147 @@
+    const institution = createFakeInstitution();
+    institution.operatingName = FAKE_INSTITUTION_NAME;
+    institution.legalOperatingName = FAKE_INSTITUTION_LEGAL_NAME;
+    await db.institution.save(institution);
+    const legalNameSearchText = "Search E2E Test institution legal";
+    const mailingAddress = institution.institutionAddress.mailingAddress;
+    const expectedSearchResult: SearchInstitutionAPIOutDTO[] = [
+      {
+        id: institution.id,
+        legalName: institution.legalOperatingName,
+        operatingName: institution.operatingName,
+        address: {
+          addressLine1: mailingAddress.addressLine1,
+          addressLine2: mailingAddress.addressLine2,
+          city: mailingAddress.city,
+          provinceState: mailingAddress.provinceState,
+          country: mailingAddress.country,
+          postalCode: mailingAddress.postalCode,
+        },
+      },
+    ];
+
+    const endpoint = `/aest/institution/search?legalName=${legalNameSearchText}`;
+    const token = await getAESTToken(AESTGroups.BusinessAdministrators);
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .then((response) => {
+        const searchResult = response.body as SearchInstitutionAPIOutDTO[];
+        expect(searchResult).toEqual(
+          expect.arrayContaining(expectedSearchResult),
+        );
+      });
+  });
+
+  it("Should return institutions when searched by legal name and operating name.", async () => {
+    // Arrange
+    const institution = createFakeInstitution();
+    institution.operatingName = FAKE_INSTITUTION_NAME;
+    institution.legalOperatingName = FAKE_INSTITUTION_LEGAL_NAME;
+    await db.institution.save(institution);
+    const searchText = "Search E2E Test institution";
+    const mailingAddress = institution.institutionAddress.mailingAddress;
+    const expectedSearchResult: SearchInstitutionAPIOutDTO[] = [
+      {
+        id: institution.id,
+        legalName: institution.legalOperatingName,
+        operatingName: institution.operatingName,
+        address: {
+          addressLine1: mailingAddress.addressLine1,
+          addressLine2: mailingAddress.addressLine2,
+          city: mailingAddress.city,
+          provinceState: mailingAddress.provinceState,
+          country: mailingAddress.country,
+          postalCode: mailingAddress.postalCode,
+        },
+      },
+    ];
+
+    const endpoint = `/aest/institution/search?legalName=${searchText}&operatingName=${searchText}`;
+    const token = await getAESTToken(AESTGroups.Operations);
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .then((response) => {
+        const searchResult = response.body as SearchInstitutionAPIOutDTO[];
+        expect(searchResult).toEqual(
+          expect.arrayContaining(expectedSearchResult),
```

Comment:
If the `legalOperatingName` is unique (a GUID for instance) we can ensure the exact result to be returned. If I am not wrong we did the same for other tests even testing the "ILIKE" also.